### ISSUE TITLE
Clustering

### ISSUE THREAD

**Issue by gedw99** [anxiousmodernman/co-chair#18]
Not sure if co-chair does this but what about clustering using boltdb raft ?
That is how consul do it but since you want a very simple system you can just use one of the boltdb raft libraries.

**Comment by anxiousmodernman** [comment]
This is a future goal. One obstacle would be the interplay between the storm ORM (which uses a boltdb fork) and any such raft library. Our data access is simple enough that we could switch to a lower-level key-value lookup without too much trouble.

**Comment by gedw99** [comment]
ah interesting. I did not know storm uses a fork of boltdb.

I guess etcd or consul then later...

Shall i close this now ?

**Comment by anxiousmodernman** [comment]
We can leave this open as a stake in the ground for the feature itself. I would be down to run experimental branches of storm that support this, or perhaps integrate things manually with a generic raft library.

**Comment by gedw99** [comment]
Yeah i saw that storm is being rearchitected but i really doubt storm will ever get clustering.
I would just run etcd
https://coreos.com/etcd/docs/latest/learning/why.html

 ! Its also much better than consul and plenty of example for using it as a low transaction DB. 

You also will get real time push, so when when a value changes it will automatically update the proxy.
Its only for low write volume but thats what this project would have, so its maybe a pretty good solution.

As it say on their web page is stored more than a few GB of data use CockroachDB.
If you choose to go with cockroach then its very easy to run.
Also there is Kallax for talking the potsresql protocol and grpc.
see here !!!
https://github.com/src-d/proteus/issues/104
Proetues allows you to do GRPC where you model from a golang structs.
Kallax allows you to build your datamodel and data base from the golang structs.
I asked if they wanted to support Kallax with cockroachdb, but it works anyway actually :)

So food for thought... I am only thinking through an easy way to get HA and fast development.
